TestColorMap.cpp

```diff
@@ -0,0 +1,61 @@
+#include "TestColorMap.h"
+
+TEST(ColorMapTest, TestGetColorPairs) {
+    auto colorPairs = colorMap.getColorPairs();
+
+    std::vector<std::pair<std::string, std::string>> expectedPairs = {
+        {"White", "Blue"}, {"White", "Orange"},
+        {"White", "Green"}, {"White", "Brown"},
+        {"White", "Slate"}, {"Red", "Blue"},
+        {"Red", "Orange"}, {"Red", "Green"},
+        {"Red", "Brown"}, {"Red", "Slate"},
+        {"Black", "Blue"}, {"Black", "Orange"},
+        {"Black", "Green"}, {"Black", "Brown"},
+        {"Black", "Slate"}, {"Yellow", "Blue"},
+        {"Yellow", "Orange"}, {"Yellow", "Green"},
+        {"Yellow", "Brown"}, {"Yellow", "Slate"},
+        {"Violet", "Blue"}, {"Violet", "Orange"},
+        {"Violet", "Green"}, {"Violet", "Brown"},
+        {"Violet", "Slate"}
+    };
+
+    EXPECT_EQ(colorPairs, expectedPairs);
+}
+
+TEST(ColorMapTest, TestFormatColorMap) {
+    ColorMap colorMap;
+    std::string result = colorMap.formatColorMap();
+    std::string expected =
+        "0 | White | Blue\n"

```

Comment:
As per the standard, the "expected" output must have pair numbers from 1 to 25, not 0 to 24. Also, for a user, the output is expected to be aligned. The column-separators are not aligned in this case.
Try to capture the user-need in the test --> so that "passing tests" will give us confidence that the user will find "the solution acceptable"

Reply:
Changed the pair numbers to 1 to 25.
And for alignment issue I have created a function to generate the expected output and then compare with generated.